src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsFrameworkCompatibility.cs

```diff
@@ -17,12 +18,29 @@ namespace NuGet.VisualStudio
     [Export(typeof(IVsFrameworkCompatibility3))]
     public class VsFrameworkCompatibility : IVsFrameworkCompatibility2, IVsFrameworkCompatibility3
     {
+        private INuGetTelemetryProvider _telemetryProvider;
+
+        [ImportingConstructor]
+        public VsFrameworkCompatibility(INuGetTelemetryProvider telemetryProvider)
+        {
+            _telemetryProvider = telemetryProvider;
+        }
+
         public IEnumerable<FrameworkName> GetNetStandardFrameworks()
         {
-            return DefaultFrameworkNameProvider
-                .Instance
-                .GetNetStandardVersions()
-                .Select(framework => new FrameworkName(framework.DotNetFrameworkName));
+            try
+            {
+                return DefaultFrameworkNameProvider
+                    .Instance
+                    .GetNetStandardVersions()
+                    .Select(framework => new FrameworkName(framework.DotNetFrameworkName))
+                    .ToList();
+            }
+            catch (Exception exception)
+            {
+                _telemetryProvider.PostFault(exception, typeof(VsFrameworkCompatibility).FullName);
```

Comment:
Wouldn't it be more convenient to have the method name somehow/somewhere?

Reply:
The method signature is `void PostFault(Exception e, string callerClassName, [CallerMemberName] string callerMemberName = null, IDictionary<string, object> extraProperties = null);`.

The `[CallerMemberName]` attribute tells the compiler that when the value is not specified, to automatally use `nameof(CurrentMethodName)`. Hence we are getting the method name, we just don't need to be explicit about it.